Feat add bf notice

cristian-ungureanu opened the pull request:
### Summary
Added the bf notice in Neve's dashboard

### Will affect the visual aspect of the product
NO


### Test instructions


- Download this version of [neve.zip](https://github.com/Codeinwp/neve/files/9911113/neve.zip)
- Check the banner in the dashboard


<!-- Issues that this pull request closes. -->
Closes Codeinwp/themeisle#1397.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

selul reviewed (approved):
Loooking good. Maybe to make things easier for testing you can attach a zip with the change?

cristian-ungureanu commented:
It took me a while to figure out the SVG as the text inside it was variable. I've changed the design a bit because the text didn't fit right. When the "less than 24 h left" text was present, the text was outside the box so I needed to make the box larger. When "x days left" was preset, the text was too short and it looked weird center aligned inside that long box without the border that I've added.
In the end, this is the result:
<img width="1034" alt="Screenshot 2022-10-28 at 17 31 37" src="https://user-images.githubusercontent.com/9929553/198649299-d3f88c8d-0786-43a6-ade3-888221f251ea.png">
<img width="1026" alt="Screenshot 2022-10-28 at 17 31 59" src="https://user-images.githubusercontent.com/9929553/198649313-266017c9-6f65-4b81-9882-ac2dc9093cc7.png">

Let me know your thoughts.

cristian-ungureanu commented:
Here are some screenshots of the dashboard with different screen resolutions:
<img width="1069" alt="Screenshot 2022-10-28 at 17 40 34" src="https://user-images.githubusercontent.com/9929553/198652688-f7bd9c6f-cefa-4688-93d5-8909862e465b.png">
<img width="1534" alt="Screenshot 2022-10-28 at 17 40 04" src="https://user-images.githubusercontent.com/9929553/198652705-344d3741-036e-4eae-abc2-6093acf36135.png">

abaicus reviewed (commented):
@cristian-ungureanu this looks good. 👍🏻 

I am not sure why we add the full SVG code to the bundle and not simply use this as an IMG tag. Is there a reason? Right now it's being bundled with the JS and increases the bundle size.

abaicus commented:
Sorry for the mix-up. I didn't notice that the days part is dynamic. It would probably be a good idea to add the top-right part as an absolute overlay over the image.

cristian-ungureanu commented:
@abaicus, the idea came from here https://github.com/Codeinwp/themeisle/issues/1397#issuecomment-1293128428. The text in the corner is variable, so we decided to do this instead of having seven images and getting another one each day. I tried to optimize the SVG as well as I could.

ineagu commented:
So this would be only in the neve dashboard?

cristian-ungureanu commented:
> So this would be only in the neve dashboard?

Yes, one in Neve's dashboard, another one in Otter's dashboard.

ineagu commented:
I cant see how this is useful, I dont think any old user visit the dashboard.

If we wanna see some results they should be sitewide and auto hide after the campaign.

Sent from my iPhone

> On Oct 28, 2022, at 16:59, cristian-ungureanu ***@***.***> wrote:
> 
> ﻿
> So this would be only in the neve dashboard?
> 
> Yes, one in Neve's dashboard, another one in Otter's dashboard.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.

abaicus commented:
> @abaicus, the idea came from here [Codeinwp/themeisle#1397 (comment)](https://github.com/Codeinwp/themeisle/issues/1397#issuecomment-1293128428). The text in the corner is variable, so we decided to do this instead of having seven images and getting another one each day. I tried to optimize the SVG as well as I could.

I don't think it's that bad if we remove this from the bundle after the promotion is done. That being said, I think it would be easier if we'd do it as an image combined with an `absolute` positioned element. The current version seems to yield different results depending on the browser. 👇

![image](https://user-images.githubusercontent.com/15010186/198667633-80f66869-f0f5-4e9e-9159-971b035f1a4b.png)

The display would also be more reliable using other browsers: 


Safari
---
![image](https://user-images.githubusercontent.com/15010186/198670245-84e9198a-34c8-4b5a-b91f-7c48eac41c67.png)

Firefox
---
![image](https://user-images.githubusercontent.com/15010186/198670408-f7041a75-6432-48e9-bbbc-ad11ee80051a.png)

cristian-ungureanu commented:
> I cant see how this is useful, I dont think any old user visit the dashboard. If we wanna see some results they should be sitewide and auto hide after the campaign.

Things might get out of hand if we do this 🙈. https://wptavern.com/black-friday-banner-gone-wrong-advertising-in-free-plugins

> I don't think it's that bad if we remove this from the bundle after the promotion is done. That being said, I think it would be easier if we'd do it as an image combined with an `absolute` positioned element. 

Thanks for the feedback, I'll follow your suggestion

ineagu commented:
@cristian-ungureanu the issue there was that it was Yoast, it was very ugly & it was not really dismissable.

Another way we can do it @selul can be a native design, but sitewide.

abaicus commented on assets/apps/dashboard/src/Components/Promotions.js:
```diff
@@ -0,0 +1,26 @@
+			{Object.keys(neveDash.promotions).map((slug, index) => {
+				return (
+					<Promotion
+						key={index}
```
It's generally a bad practice to use the index of an array item as a key. We can use the `slug` here.

abaicus commented on assets/apps/dashboard/src/Components/Content/PromotionsContent.js:
```diff
@@ -0,0 +1,5 @@
+const promotions = {
+	bf22: require('../../../../../img/dashboard/bf22.png'),
+};
+
+export default promotions;
```
Is there any reason for using `require` instead of importing the image here? 
```suggestion
import bf22 from '../../../../../img/dashboard/bf22.png';

export default {
	bf22,
};
```

abaicus reviewed (changes requested):
@cristian-ungureanu, here are a few things in addition to the ones mentioned on the code:

- we can keep the banner as a .svg file so it doesn't get blurred;
- let's drop the `PromotionsContent.js` file and localize the image URL. This way, we keep the data coming in from a single source. We can also simplify the Promotion component, dropping the `slug` prop which is only used to retrieve data from the promotions object;

cristian-ungureanu commented:
Hey @abaicus, I've added the customizer and the sitewide dashboard notices. Can you have another look, please? 🙏

abaicus reviewed (commented):
@cristian-ungureanu, this generally looks ok. Left some things below.

- in my opinion, the customizer banner can be SVG as well;
- the `should_show_bf` is sitewide and not tied to the dashboard anymore, so I'm not sure why we added it in the Main dashboard class. Maybe we can move it to a trait or something more general. (but it's not a big deal as long as we remove it once BF passes);

irinelenache commented:
@cristian-ungureanu Tested the notice and here are my findings:

- [ ] The Neve logo is not visible in the dashboard notice https://vertis.d.pr/i/9RMMgi
- [ ] The banner is visible in the starter sites section until i activate TPC ([before](https://vertis.d.pr/i/CE5oZc) activating - [after](https://vertis.d.pr/i/kaVfq0) activating). I think it's not a problem but is worth mentioning to double check
- [ ] The time left section doesn't have the dashed border as in the screenshots https://vertis.d.pr/i/xQuWhk
- [ ] If i have Neve pro activated but have no license for it, the banner won't be visible

abaicus commented:
@cristian-ungureanu Can we use the [same SVG logo](https://github.com/Codeinwp/neve/blob/master/assets/img/dashboard/logo.svg) from the dashboard for the notice like I mentioned [here](https://github.com/Codeinwp/neve/pull/3673#discussion_r1010372251)?

cristian-ungureanu commented:
Thanks for checking, @irinelenache 
1. Fixed!
2. Not a problem
3. The original design was without a dashed border. That's what I improvised to fix the alignment in a former implementation but it's not required anymore.
4. Fixed!
